data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/beacon/SwaggerIntegrationTest.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.beaconrestapi.beacon;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.Resources;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import okhttp3.Response;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import tech.pegasys.teku.beaconrestapi.AbstractDataBackedRestAPIIntegrationTest;
+
+public class SwaggerIntegrationTest extends AbstractDataBackedRestAPIIntegrationTest {
+  private final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  private JsonNode currentJsonNodes;
+
+  @BeforeEach
+  public void setup() throws IOException {
+    startRestAPIAtGenesis();
+    final Response response = get();
+    currentJsonNodes = OBJECT_MAPPER.readTree(response.body().string());
+  }
+
+  @Test
+  void schemaObjects_shouldBeConsistent(@TempDir final Path tempDir) throws IOException {
+    final JsonNode schemas = currentJsonNodes.findPath("components").findPath("schemas");
+
+    List<String> fieldNames = ImmutableList.copyOf(schemas.fieldNames());
+    for (String current : fieldNames) {
+      checkStructureIsConsistent("schema", tempDir, current, schemas.path(current));
+    }
+    checkAllExpectedFilesExist(tempDir, "schema");
+  }
+
+  @Test
+  void paths_shouldBeConsistent(@TempDir final Path tempDir) throws IOException {
+    final JsonNode paths = currentJsonNodes.findPath("paths");
+
+    List<String> fieldNames = ImmutableList.copyOf(paths.fieldNames());
+    for (String current : fieldNames) {
+      checkStructureIsConsistent("paths", tempDir, current, paths.path(current));
+    }
+
+    checkAllExpectedFilesExist(tempDir, "paths");
+  }
+
+  private void checkAllExpectedFilesExist(final Path tempDir, final String schema)
+      throws IOException {
+    final List<String> expectedFileListing =
+        Arrays.asList(
+            Resources.toString(
+                    Resources.getResource(SwaggerIntegrationTest.class, schema + "_listing.txt"),
+                    UTF_8)
+                .split("\n"));
+    for (String file : expectedFileListing) {
+      assertThat(tempDir.resolve(file).toFile()).exists();
+    }
```

Comment:
I suspect we should fail if we have extra items as well. That way if new types are added we are "reminded" to add the expected schema so this test covers them for future regressions.

I suspect that just means creating a set of the expected files and a set of the files that do exist and asserting they're equal.

Reply:
i'll double check, but this is the pass back through, so i think we're good... will look into the scenarios

Reply:
Delete one of the files and check the test fails, if so - we're good. :). I think you're probably right thinking about it.

Reply:
`java.lang.IllegalArgumentException: resource schema/AggregateAndProof.json relative to tech.pegasys.teku.beaconrestapi.beacon.SwaggerIntegrationTest not found`
:)